spec/fog/compute/xen_server/models/pif_spec.rb

```diff
@@ -115,4 +115,11 @@ def self.read_identity
   it "shouldn't require attributes before save" do
     pif_class.require_before_save.must_equal([])
   end
-end
\ No newline at end of file
+
+  it 'should define methods' do
+    methods = pif_class.instance_methods(false)
+    pif_class::PIF_METHODS.each do |method|
+      methods.include?(method).must_equal(true)
+    end
+  end
```

Comment:
You know when the code is good when the test can be as simple as that. Great refactoring including tests. =)